File: web/src/main/java/by/sorface/passport/web/dao/sql/RoleRepository.java
package by.sorface.passport.web.dao.sql;

import by.sorface.passport.web.dao.models.RoleEntity;
import org.springframework.stereotype.Repository;

import java.util.Optional;

@Repository
public interface RoleRepository extends BaseRepository<RoleEntity> {

    Optional<RoleEntity> findByValueIgnoreCase(String value);

}

File: web/src/main/kotlin/by/sorface/passport/web/dao/sql/RoleRepository.kt
package by.sorface.passport.web.dao.sql

import by.sorface.passport.web.dao.models.RoleEntity
import org.springframework.stereotype.Repository
import java.util.*

@Repository
interface RoleRepository : BaseRepository<RoleEntity?> {

    fun findByValueIgnoreCase(value: String?): Optional<RoleEntity?>

}
